dspy/clients/lm.py

```diff
@@ -60,9 +87,76 @@ def __call__(self, prompt=None, messages=None, **kwargs):
         
         return outputs
 
+    def copy(self, **kwargs):
```

Comment:
copy with updated args is strange, essentially here we are just calling the constructor. I would rather provide a function `get_config` to get all the args, and pass to constructor.

Reply:
I agree but lm.copy is an established pattern. This is actually a duplicate function from the copy function already in lm. Removing it

Reply:
This is actually a pretty established pattern iirc, both inside DSPy and in other libraries. It's a non-inplace way to modify an object (getting a copy with overwrites).